appshared/src/main/java/io/reark/rxgithubapp/shared/viewmodels/RepositoryViewModel.java

```diff
@@ -57,19 +57,18 @@ public RepositoryViewModel(@NonNull final GetUserSettings getUserSettings,
     }
 
     @Override
-    public void subscribeToDataStoreInternal(@NonNull final CompositeSubscription compositeSubscription) {
-        checkNotNull(compositeSubscription);
-
-        compositeSubscription.add(
+    public void subscribeToDataStoreInternal(@NonNull final CompositeDisposable compositeDisposable) {
+        checkNotNull(compositeDisposable);
+                compositeDisposable.add(
                 getUserSettings.call()
                         .map(UserSettings::getSelectedRepositoryId)
                         .switchMap(fetchAndGetGitHubRepository::call)
-                        .subscribe(repository)
+                        .subscribe(repository::onNext)
```

Comment:
Here is the reason why the `testRepositoryViewModelFetchesValidGitHubRepository` fails.
You are passing now only the `onNext` and in test it waits for completed event as well